node/src/components/linear_chain_synchronizer/event.rs

```diff
@@ -0,0 +1,56 @@
+use std::fmt::{self, Display, Formatter};
+
+use serde::Serialize;
+
+use casper_execution_engine::core::engine_state::{self, genesis::GenesisSuccess, UpgradeSuccess};
+
+use super::Error;
+use crate::{
+    contract_runtime::{BlockAndExecutionEffects, BlockExecutionError},
+    types::{BlockHash, BlockHeader},
+};
+
+#[derive(Debug, Serialize)]
+pub(crate) enum Event {
+    /// The result of getting the highest block from storage.
+    HighestBlockHash(Option<BlockHash>),
+    /// The result of the fast sync task.
+    SyncResult(Result<BlockHeader, Error>),
```

Comment:
I'll probably go the other way and remove the `fast` from the doc comment.  I think it's clearer to just stick to calling this "chain sync" throughout (as per @afck feedback) rather than a mixture of "linear chain sync" and "fast sync".

Reply:
Fixed in [`2d3671c2`](https://github.com/casper-network/casper-node/pull/2553/commits/2d3671c2e1cf45578eb1b096edba71d4fa347994).